File: Nunit.Test/Examples/Inheritance/GetType_or_TypeOf.cs
using System;
using NUnit.Framework;

namespace Examples.Examples.Inheritance
{
	public class GetType_or_TypeOf
	{
		public class Animal { }
		public class Dog : Animal { }

		[Test]
		public void Checkes()
		{
			var spot = new Dog();

			Assert.Multiple(() => {
				Assert.That(spot.GetType() == typeof(Animal), Is.False);
				Assert.That((spot is Animal), Is.True);
				Assert.That(spot.GetType() == typeof(Dog), Is.True);
				Assert.That(spot is Dog, Is.True);
			});

		}

	}
}

File: NUnit.vb.dotnet35.Test/Examples/GetType_or_TypeOf.vb
Imports NUnit.Framework

Namespace Examples
	Public Class GetType_or_TypeOf

		Public Class Animal

		End Class
		Public Class Dog
			Inherits Animal
		End Class

		<Test>
		Public Sub CheckType()
			Dim spot As New Dog()

			Assert.Multiple(Sub()
								Assert.That(spot.GetType() Is GetType(Animal), [Is].False)
								Assert.That(TypeOf spot Is Animal, [Is].True)
								Assert.That(spot.GetType() Is GetType(Dog), [Is].True)
								Assert.That(TypeOf spot Is Dog, [Is].True)
							End Sub)
		End Sub
	End Class
End Namespace
